### PR TITLE
Review request for Microsoft.ContainerService to add version 2022-06-01

### PR DESCRIPTION
**Pull request by FumingZhang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
    Deployed to global cloud by 07/07, sov cloud by 07/12.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
    07/21.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by FumingZhang** [comment]
Dear review board, I would like to clarify the failures in the CI checks.

For [SDK azure-powershell](https://github.com/Azure/azure-rest-api-specs/pull/19848/checks?check_run_id=7423171988), it happened in our previous PRs like stable/[04-01](https://github.com/Azure/azure-rest-api-specs/pull/18991) and preview/[06-02-preview](https://github.com/Azure/azure-rest-api-specs/pull/19653). This used to be optional and is enabled recently. Confirmed with owner @dolauli, this would not work for some modules and could be ignored.

For [SDK azure-sdk-for-net](https://github.com/Azure/azure-rest-api-specs/pull/19848/checks?check_run_id=7423171283), it also happened in our previous PRs like stable/[04-01](https://github.com/Azure/azure-rest-api-specs/pull/18991) and preview/[06-02-preview](https://github.com/Azure/azure-rest-api-specs/pull/19653). I suppose this should be a known issue.

For [Swagger Avocado](https://github.com/Azure/azure-rest-api-specs/pull/19848/checks?check_run_id=7423149690), it happened in our previous PR stable/[04-01](https://github.com/Azure/azure-rest-api-specs/pull/18991). I've listed the detailed explanation for each error in [comment](https://github.com/Azure/azure-rest-api-specs/pull/18686#issuecomment-1099823292). These errors started appearing after [this update](https://github.com/Azure/avocado/pull/78) of Avocado. Would really appreciate if some mechanism could be provided to disable the alarms, cc owner @ruowan.

For [Breaking Change(Cross-Version)](https://github.com/Azure/azure-rest-api-specs/pull/19653/checks?check_run_id=7173329826) and [Swagger SDK Breaking Change Tracking](https://github.com/Azure/azure-rest-api-specs/pull/19653/checks?check_run_id=7173330296), it's caused by rebranding of Microsoft Defender. The modification has been added to our **previous preview** API ([2022-05-02-preview](https://github.com/Azure/azure-rest-api-specs/pull/19262)) and is about to be introduced into this stable API 2022-06-01. Feature owner has got the approval from Jeffrey, see [comment](https://github.com/Azure/azure-rest-api-specs/pull/19262#issuecomment-1143126662).

**Comment by rkmanda** [comment]
>           "x-ms-identifiers": [],

is there not any actual unique property or combination of proprties that are unique in the array

---
Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json:4148 in 84159c9. [](commit_id = 84159c97605238946622563dd34fa5d10c412cc0, deletion_comment = False)

**Comment by rkmanda** [comment]
>           "x-ms-identifiers": [],

pl check everywhere 

---
In reply to: [1190801564](https://github.com/Azure/azure-rest-api-specs/pull/19848#issuecomment-1190801564) [](http://example.com/codeflow?ancestors=1190801564)

---
Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json:4148 in 06d72d6. [](commit_id = 06d72d6ac9978b3a9488d6a838d64eedd2d8acf6, deletion_comment = False)

**Review by jianyexi — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/examples/ManagedClustersCreate_SecurityProfile.json
Comment (palma21) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/examples/ManagedClustersCreate_SecurityProfile.json:244]: @matthchr @FumingZhang was this approved? If not this should be removed

**Comment by matthchr** [comment]
I approved [here](https://dev.azure.com/msazure/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/243159/MicrosoftDefender-rebranding) but I don't see others approvals.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/examples/ManagedClustersCreate_SecurityProfile.json
Comment (FumingZhang) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/examples/ManagedClustersCreate_SecurityProfile.json:244]: This PR has been merged. I'll remove this wrong example and add the missing examples back in the following API version.

I've opened a work item to track this.

Comment (FumingZhang) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/examples/ManagedClustersCreate_SecurityProfile.json:244]: I took a closer look, this example should be fine? I suppose it is following the data model after the rebranding of defender. @palma21

**Pull request merged**
